substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2239,13 +2238,18 @@ pub mod pallet {
 			let member_account = T::Lookup::lookup(member_account)?;
 			let mut member =
 				PoolMembers::<T>::get(&member_account).ok_or(Error::<T>::PoolMemberNotFound)?;
-			let current_era = T::Staking::current_era();
+			let current_era = T::StakeAdapter::current_era();
 
 			let bonded_pool = BondedPool::<T>::get(member.pool_id)
 				.defensive_ok_or::<Error<T>>(DefensiveError::PoolNotFound.into())?;
 			let mut sub_pools =
 				SubPoolsStorage::<T>::get(member.pool_id).ok_or(Error::<T>::SubPoolsNotFound)?;
 
+			if bonded_pool.has_pending_slash() {
+				// apply slash if any before withdraw.
```

Comment:
Option 1 from above continued: we could do `<frame_system::Pallet<T>>::register_extra_weight_unchecked(T::WeightInfo::apply_slash())` here in this `if`.

Reply:
The `has_pending_slash` check is duplicated in `do_apply_slash`.